src/uproot/_util.py

```diff
@@ -570,6 +570,14 @@ def awkward_form(model, file, context):
                 _primitive_awkward_form[model] = awkward.forms.from_json('"float32"')
             elif model == numpy.dtype(numpy.float64):
                 _primitive_awkward_form[model] = awkward.forms.from_json('"float64"')
+            elif model.fields is not None:
+                fields = []
+                contents = []
+                for field, (dtype, _) in model.fields.items():
+                    fields.append(field)
+                    contents.append(awkward_form(dtype, file, context))
+                # directly return; don't cache RecordForms in _primitive_awkward_form
+                return awkward.forms.RecordForm(contents, fields)
```

Comment:
This is unrelated to the main purpose of the PR, but calling the `awkward_forth` method at this point in the code revealed that the leaf-list case isn't handled. Somehow, `test_0033-more-interpretations-2.py::test_leaflist_awkward` was passing before because something must have figured out that it needs a `RecordArray` without calling `awkward_forth`. (I'm not sure which code path that is, but now, at least, this is consistent with that.)